sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/RetryContextOnDiagnosticTest.java

```diff
@@ -0,0 +1,239 @@
+import java.util.function.Function;
+
+public class RetryContextOnDiagnosticTest extends TestSuiteBase {
+    private final static String exceptionText = "TestException";
+    private final static String responseText = "TestResponse";
+    private final static String itemSelfLink = "dbs/TestDB/colls/TestColl/docs/TestDoc";
+    private final static String containerSelfLink = "dbs/TestDB/colls/TestColl";
+
+    private IRetryPolicy retryPolicy;
+    private Callable<Mono<StoreResponse>> callbackMethod;
+    private Function<Quadruple<Boolean, Boolean, Duration, Integer>, Mono<StoreResponse>> parameterizedCallbackMethod;
+    private Function<Quadruple<Boolean, Boolean, Duration, Integer>, Mono<StoreResponse>> inBackoffAlternateCallbackMethod;
+    private RxDocumentServiceRequest serviceRequest;
+
+    @BeforeClass(groups = "unit")
+    public void RetryContextTest() {
+        callbackMethod = Mockito.mock(Callable.class);
+        parameterizedCallbackMethod = Mockito.mock(Function.class);
+        inBackoffAlternateCallbackMethod = Mockito.mock(Function.class);
+    }
+
+    @Test(groups = {"simple"})
+    public void backoffRetryUtilityExecuteRetryTest() throws Exception {
+        serviceRequest = RxDocumentServiceRequest.create(OperationType.Read, ResourceType.Document);
+        retryPolicy = new TestRetryPolicy();
+        CosmosClientException exception = new CosmosClientException(410, exceptionText);
+        Mockito.when(callbackMethod.call()).thenThrow(exception, exception, exception, exception, exception)
+            .thenReturn(Mono.just(new StoreResponse(200, new ArrayList<>(), responseText)));
+        Mono<StoreResponse> monoResponse = BackoffRetryUtility.executeRetry(callbackMethod, retryPolicy);
+        StoreResponse response = validateSuccess(monoResponse);
+
+        Assertions.assertThat(response.getResponseBody()).isEqualTo(responseText);
+        Assertions.assertThat(retryPolicy.getRetryCount()).isEqualTo(5);
+        Assertions.assertThat(retryPolicy.getStatusAndSubStatusCodes().size()).isEqualTo(retryPolicy.getRetryCount());
+    }
+
+    @Test(groups = {"simple"})
+    public void backoffRetryUtilityExecuteRetryWithFailureTest() throws Exception {
+        serviceRequest = RxDocumentServiceRequest.create(OperationType.Read, ResourceType.Document);
+        retryPolicy = new TestRetryPolicy();
+        CosmosClientException exception = new CosmosClientException(410, exceptionText);
+        Mockito.when(callbackMethod.call()).thenThrow(exception);
+        ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
+        executor.schedule(() -> {
+            ((TestRetryPolicy) retryPolicy).noRetry = true;
+        }, 10, TimeUnit.SECONDS);
+        Mono<StoreResponse> monoResponse = BackoffRetryUtility.executeRetry(callbackMethod, retryPolicy);
+        validateFailure(monoResponse);
+
+        Assertions.assertThat(retryPolicy.getRetryCount()).isGreaterThanOrEqualTo(5);
```

Comment:
please use static import for `assertThat` so that similar to other tests instead of `Assertions.assertThat` you do `assertThat` here and in other tests.

Reply:
Done